gridstatus/ercot.py

```diff
@@ -3351,3 +3353,56 @@ def _weather_zone_column_name_order(self):
             "Southern",
             "West",
         ]
+
+    @support_date_range(frequency=None)
+    def get_indicative_lmp_by_settlement_point(
+        self,
+        date: str | pd.Timestamp | tuple[pd.Timestamp, pd.Timestamp],
+        end: str | pd.Timestamp | tuple[pd.Timestamp, pd.Timestamp] | None = None,
+        verbose: bool = False,
+    ) -> pd.DataFrame:
+        if date == "latest":
+            self.get_indicative_lmp_by_settlement_point(date="today")
+        if not end:
+            end = date + pd.DateOffset(days=1)
+
+        docs = self._get_documents(
+            report_type_id=ERCOT_INDICATIVE_LMP_BY_SETTLEMENT_POINT_RTID,
+            date=date,
+            extension="csv",
+            published_before=end,
+            published_after=date,
+            verbose=verbose,
+        )
+        df = self.read_docs(docs, parse=False, verbose=verbose)
+        columns_to_rename = {
+            "RepeatedHourFlag": "DSTFlag",
+            "IntervalId": "Interval Id",
+            "IntervalEnding": "Interval End",
+            "IntervalRepeatedHourFlag": "Interval Repeated Hour Flag",
+            "SettlementPoint": "Location",
+            "SettlementPointType": "Location Type",
+            "LMP": "LMP",
+        }
+        df.rename(columns=columns_to_rename, inplace=True)
+        df["RTDTimestamp"] = pd.to_datetime(df["RTDTimestamp"]).dt.tz_localize(
+            self.default_timezone,
+        )
+        df["Interval End"] = pd.to_datetime(df["Interval End"]).dt.tz_localize(
+            self.default_timezone,
+        )
+
+        df["Interval Start"] = df["Interval End"] - pd.Timedelta(minutes=5)
+        df = df.sort_values("Interval Start").reset_index(drop=True)
+        return df[
+            [
+                "Interval Start",
+                "Interval End",
+                "RTDTimestamp",
+                "Interval Id",
+                "Interval Repeated Hour Flag",
+                "Location",
+                "Location Type",
+                "LMP",
+            ]
+        ]
```

Comment:
I think we can move this to a utility function (something like `_handle_indicative_lmp_by_settlement_point`) in this class then call that method here and in `ErcotAPI`.

Reply:
sounds good, done.